eng/pipelines/templates/stages/archetype-java-release.yml

```diff
@@ -388,47 +390,48 @@ stages:
-                Write-Host "'setting.xml' exists. Overwriting the file to support multiple repositories."
-              }
-              Copy-Item "./eng/repo-docs/docms/daily.update.setting.xml" -Destination "~/.m2/settings.xml"
-            displayName: 'Configure mvn'
-            workingDirectory: $(Build.SourcesDirectory)
-          - template: /eng/common/pipelines/templates/steps/update-docsms-metadata.yml
-            parameters:
-              PackageInfoLocations:
-                - ${{ each artifact in parameters.Artifacts }}:
-                  - ${{if ne(artifact.skipPublishDocMs, 'true')}}:
-                    - $(Pipeline.Workspace)/${{parameters.ArtifactName}}/PackageInfo/${{artifact.name}}.json
-              WorkingDirectory: $(System.DefaultWorkingDirectory)
-              TargetDocRepoOwner: "azure-sdk"
-              TargetDocRepoName: ${{parameters.TargetDocRepoName}}
-              Language: 'java'
-              DailyDocsBuild: true
-              SparseCheckoutPaths:
-                - docs-ref-services/
-                - metadata/
-              DocValidationImageId: "$(DocValidationImageId)"
+      - ${{if ne(parameters.SkipServicePublishDocsMs, 'true')}}:
+        - job: PublishDocsToNightlyBranch
+          dependsOn: PublishPackages
+          condition: or(eq(variables['SetDevVersion'], 'true'), and(eq(variables['Build.Reason'],'Schedule'), eq(variables['System.TeamProject'], 'internal')))
+          pool:
+            name: azsdk-pool-mms-ubuntu-2004-general
+            vmImage: MMSUbuntu20.04
+          variables:
+            DocValidationImageId: azuresdkimages.azurecr.io/javarefautocr:latest
+          steps:
+            - download: current
+              displayName: 'Download Artifact: ${{parameters.ArtifactName}}'
+              artifact: ${{parameters.ArtifactName}}
+            - pwsh: |
+                Get-ChildItem -Recurse $(Pipeline.Workspace)/${{parameters.ArtifactName}}/
+              displayName: Show visible artifacts
+            # Docs daily updates is supposed to download packages from public feed repository, so we have to specify additional repositories in a POM or the profile.
+            # Here is maven documentation: https://maven.apache.org/guides/mini/guide-multiple-repositories.html
+            - powershell: |
+                # Linux mvn `setting.xml` is sitting under path `~/.m2/setting.xml`
+                Get-Command mvn
+                if (!(Test-Path '~/.m2/')) {
+                  mkdir ~/.m2/
+                }
+                if (Test-Path '~/.m2/setting.xml') {
+                  Write-Host "'setting.xml' exists. Overwriting the file to support multiple repositories."
+                }
+                Copy-Item "./eng/repo-docs/docms/daily.update.setting.xml" -Destination "~/.m2/settings.xml"
+              displayName: 'Configure mvn'
+              workingDirectory: $(Build.SourcesDirectory)
+            - ${{ each artifact in parameters.Artifacts }}:
```

Comment:
Did you intentionally want to call the update-docms-metadata.yml once per-artifact now? If this scenario works doesn't it also solve the larger problem such that we don't need to add the new SkipServicePublishDocMs parameter?

Reply:
One thing to consider is if we loop at this level I suspect it might end up with multiple commits to the docs repo one per package as opposed to one per service. I'm not sure if that matters but something to consider.

Reply:
The reason I am changing this is we currently block git push due to one bad artifact. By having this, we are able to unblock the successful ones to update.